src/build.jl

```diff
@@ -0,0 +1,103 @@
+# This file generated the Markdown documentation files.
+
+# The @file macro generates the documentation for a particular file
+# where the {{method1, methods2}} includes the documentation for each method
+# via the `buildwriter` function.
+
+# Currently this prints the methodtable followed by the docstring.
+
+macro file(fn, md_string, m::Module) buildfile(fn, md_string, m) end
+
+function files(file_names, m::Module; doc_dir="doc", gen_dir="_generated")
+    eval(Expr(:toplevel, quote using $(symbol(m)) end))
+    dd = joinpath(Pkg.dir(string(m)), doc_dir)
+    for file_name in file_names
+        eval(quote
+            @file $(joinpath(dd, gen_dir, file_name)) $(open(readall, joinpath(dd, file_name))) $m
+        end)
+    end
+end
+
+buildfile(t, s::AbstractString, m::Module) = buildfile(t, Expr(:string, s), m)
+
+buildfile(target, source::Expr, m::Module) = quote
+    open(joinpath(dirname(@__FILE__), $(esc(target))), "w") do file
+        println(" - '$($(esc(target)))'")
+        println(file, "<!-- AUTOGENERATED. See 'doc/build.jl' for source. -->")
+        $(Expr(:block, [buildwriter(arg, m) for arg in source.args]...))
+    end
+end
+
+buildwriter(ex::Expr, m::Module) = :(print(file, $(esc(ex))))
+
+buildwriter(t::AbstractString, m::Module) = Expr(:block,
+    [buildwriter(p, iseven(n), m) for (n, p) in enumerate(split(t, r"{{|}}"))]...
+)
+
+buildwriter(part, isdef, m) = isdef ?
+    begin
+        parts = Expr(:vect, [:(($(parse(p))), Base.Docs.@doc($(parse(p)))) for p in filter(x -> strip(x) != "", split(part, r"\s*[, \n]\s*"))]...)
```

Comment:
It's unclear why I added this (the Base.Docs), it appeared to be necessary to fix locally, but it's obviously not what we want! Presumably there's a better way!